### PR TITLE
support for ng-disabled #27

### PR DESCRIPTION
**Pull request by zivoradmilekic**

### CODE REVIEW

File: src/js/expansionPanel.directive.js [hunk 1]
```diff
@@ -102,15 +102,17 @@ function expansionPanelDirective() {
...
 
-      if (isDisabled === true) {
-        $element.attr('tabindex', '-1');
-      } else {
-        $element.attr('tabindex', '0');
-      }
-    });
+	setTabIndex();
+
+	$scope.$watch($attrs.ngDisabled, function(disabled) {
```
Comment (B-3PO) [inline src/js/expansionPanel.directive.js]: You could shorten the entire thing to just this.
Remove you initial ngDisabled check
Remove your setTabIndex Function

``` javascript
if ($attrs.ngDisabled !== undefined) {
  scope.$watch($attrs.ngDisabled, function(value) {
    isDiabled = value;
    $element.attr('tabindex', isDiabled ? -1 : 0);
  });
}
```

File: src/js/expansionPanel.directive.js [hunk 2]
```diff
@@ -102,15 +102,16 @@ function expansionPanelDirective() {
...
-      isDisabled = (typeof disabled === 'string' && disabled !== 'false') ? true : false;
+	isDisabled = (typeof $attrs.disabled === 'string' && $attrs.disabled !== 'false') ? true : false;
 
-      if (isDisabled === true) {
-        $element.attr('tabindex', '-1');
-      } else {
-        $element.attr('tabindex', '0');
-      }
-    });
+	$element.attr('tabindex', isDisabled ? -1 : 0);
```
Comment (B-3PO) [inline src/js/expansionPanel.directive.js]: the watcher will fire on startup so all you should need is a `if statement` wrapping the watcher. We do not want the watcher running if no `ngDisabled` attr is present. You should be able to get rid of lines 105 and 107

**Comment by B-3PO** [comment]
Can you fix the tabbing to be 2 spaces.

**Comment by zivoradmilekic** [comment]
I hope now everything is OK and ready for merge?!

**Comment by B-3PO** [comment]
There are a couple of issues. Once these are taken care of i will handle updating the version number and releasing the new version.
- we should only create a watcher if there is an `ng-disabled` attribute added
- The watcher will fire with an initial value, so we don't need to check it when `ng-disabled` is not present.
- We want to remove the is undefined check because undefined is a valid value that will be regarded as false.
- If you introduce new code to fix a bug you should be writing tests to prove your changes are correct and the bug cannot happen again. 
- Could you remove your temp changes to the docs app so the pubic docs page stays the same

Here is my suggestion on how to write to code so the watcher does not need to run when ngDisabled does not exist. I also updated my "if" check to be more readable

``` javascript
    if ($attrs.ngDisabled !== undefined) {
      $scope.$watch($attrs.ngDisabled, function(value) {
        isDisabled = value;
        $element.attr('tabindex', isDisabled ? -1 : 0);
      });
    } else if ($attrs.disabled !== undefined) {
      isDisabled = ($attrs.disabled !== undefined && $attrs.disabled !== 'false' && $attrs.disabled !== false);
      $element.attr('tabindex', isDisabled ? -1 : 0);
    }
```

### `typeof value !== 'undefined'`:

The check above is used to tell you if a variable has been declared. In the case of a functions parameters, they are always defaulted to `undefined`. Because of this, you only need to do a simple check `param !== undefined`.

Here is some code you can run that shows when you need to use this check
Test 1

``` javascript
// test one show the error
console.log(typeof temp === 'undefined'); // true
console.log(temp === undefined); // Error
```

Test 2

``` javascript
// this code shows you that declaring the var fixes the error
var temp;
console.log(typeof temp === 'undefined'); // true
console.log(temp === undefined); // true
```

Test 3

``` javascript
// this code shows you that a param that was not passed in will not throw an error
function testFunc(param) {
  console.log(param === undefined); // true
}
testFunc();
```

**Comment by B-3PO** [comment]
Closing this pr in-place of this one https://github.com/B-3PO/angular-material-expansion-panel/pull/29

**Pull request closed without merging**